src/snowflake/snowpark/_internal/analyzer/select_statement.py

```diff
@@ -779,6 +779,31 @@ def projection_in_str(self) -> str:
             )
         return self._projection_in_str
 
+    def with_subqueries(self, subquery_plans: List[SnowflakePlan]) -> "SelectStatement":
+        """Update pre-actions, post-actions and schema to capture necessary subquery_plans
+        encountered during plan resolution."""
+        for plan in subquery_plans:
+            for query in plan.queries[:-1]:
+                if self.pre_actions is None:
+                    self.pre_actions = []  # pragma: no cover
+                if query not in self.pre_actions:
+                    self.pre_actions.append(query)
+            for query in plan.post_actions:
+                if self.post_actions is None:
+                    self.post_actions = []  # pragma: no cover
+                if query not in self.post_actions:
+                    self.post_actions.append(query)
+
+            if (self._schema_query is not None) and (plan.schema_query is not None):
+                self._schema_query = self._schema_query.replace(
+                    plan.queries[-1].sql, plan.schema_query
+                )
+
+        if self._snowflake_plan is not None:
```

Comment:
i don't think you need to do this, all snowflake_plan should be handled during resolve right before you call 
```
if isinstance(logical_plan, SelectStatement):
               logical_plan.with_subqueries(self.subquery_plans)
```

Reply:
It does not.

resolve() calls do_resolve()
in do_resolve(), we only recursively call resolve again for `child in logical_plan.children`. Since for selectables, `logical_plan.children = []`, it we don't call it for the underlying `SnowflakePlan`

Reply:
for `SnowflakePlan`, `with_subqueries` does not do an in-place replacement. It creates a new `SnowflakePlan` with updated values. As a result, first level of call after applying `with_subqueries` works fine but the next levels pull snowflake_plan from `Selectable._snowflake_plan`. Since is it already resolved, it does not populate `self.subquery_plans` inside analyzer. Therefore, we need this to update `_snowflake_plan` for the selectable.